File: benchmarks/pattern/match.js
'use strict';

const Pattern = require('../../lib/router/pattern');

const pattern = new Pattern('/foo/:bar/baz');
for (let i = 0; i < 10000000; i++) {
  pattern.match('/test/bar/baz', {isEndpoint: true});
  pattern.match('/foo/bar/baz', {isEndpoint: true});
}

File: benchmarks/pattern/match.pl
use Mojo::Base -strict;

use Mojolicious::Routes::Pattern;

my $pattern = Mojolicious::Routes::Pattern->new('/foo/:bar/baz');
for (my $i = 0; $i < 10000000; $i++) {
  $pattern->match('/test/bar/baz', 1);
  $pattern->match('/foo/bar/baz',  1);
}
